src/core/ext/xds/xds_client_grpc.cc

```diff
@@ -313,10 +317,10 @@ GrpcXdsClient::GrpcXdsClient(
               .certificate_providers())),
       stats_plugin_group_(GetStatsPluginGroupForKey(key_)),
       registered_metric_callback_(stats_plugin_group_.RegisterCallback(
-          [this](CallbackMetricReporter& reporter) {
+          [this](CallbackMetricReporterWrapper& reporter) {
             ReportCallbackMetrics(reporter);
           },
-          {kMetricConnected, kMetricResources})) {}
+          {kMetricConnected.convert(), kMetricResources.convert()})) {}
```

Comment:
The need to call `convert()` here is really ugly.  I think this is a symptom of the fact that we have two different representations for each handle.  I have a suggestion elsewhere that should hopefully avoid the need to have two different types in the first place, but if we do wind up having a second type for some reason, I think it needs to be completely hidden within the API, not exposed to callers like this.

Reply:
No need of this either, it just gets implicitly converted to its base type.